kong/clustering/control_plane.lua

```diff
@@ -357,30 +357,31 @@ function _M:handle_cp_websocket(cert)
       if typ == "close" then
         ngx_log(ngx_DEBUG, _log_prefix, "received close frame from data plane", log_suffix)
         return
-      end
 
-      if not data then
-        return nil, "did not receive ping frame from data plane"
+      elseif typ == "ping" then
+        ngx_log(ngx_DEBUG, _log_prefix, "received ping frame from data plane", log_suffix)
 
-      elseif #data ~= 32 then
-        return nil, "received a ping frame from the data plane with an invalid"
-                 .. " hash: '" .. tostring(data) .. "'"
-      end
+        if not data then
+          return nil, "did not receive ping frame from data plane"
 
-      -- dps only send pings
-      if typ ~= "ping" then
-        return nil, "invalid websocket frame received from data plane: " .. typ
-      end
+        elseif #data ~= 32 then
+          return nil, "received a ping frame from the data plane with an invalid"
+                   .. " hash: '" .. tostring(data) .. "'"
+        end
 
-      ngx_log(ngx_DEBUG, _log_prefix, "received ping frame from data plane", log_suffix)
+        config_hash = data
+        last_seen = ngx_time()
+        update_sync_status()
 
-      config_hash = data
-      last_seen = ngx_time()
-      update_sync_status()
+        -- queue PONG to avoid races
+        table_insert(queue, PONG_TYPE)
+        queue.post()
 
-      -- queue PONG to avoid races
-      table_insert(queue, PONG_TYPE)
-      queue.post()
+      -- dps only send pings
+      else
+        ngx_log(ngx_DEBUG, _log_prefix, data, log_suffix)
```

Comment:
The control flow in `kong.clustering.control_plane` is intended to be the same as before, but we log any unexpected payload now before failing the connection. This means that if we remove the `konnect_mode` feature gate, the Lua CP will log the error from the DP.

If we decide to keep this PR as is (read: Konnect-only), this code change can be reverted if desired.

Reply:
update: the code change was reverted, as the feature is going to remain konnect-only for the purposes of this PR